cpp/src/Slice/Preprocessor.cpp

```diff
@@ -161,25 +149,14 @@ namespace
             args.push_back(os.str());
         }
 
-        copy(extraArgs.begin(), extraArgs.end(), back_inserter(args));
+        args.push_back(languageArg);
         args.push_back(fileName);
         return args;
     }
 }
 
 FILE*
-Slice::Preprocessor::preprocess(bool keepComments, const string& extraArg)
-{
-    vector<string> args;
-    if (!extraArg.empty())
-    {
-        args.push_back(extraArg);
-    }
-    return preprocess(keepComments, args);
-}
-
-FILE*
-Slice::Preprocessor::preprocess(bool keepComments, const vector<string>& extraArgs)
```

Comment:
Again, the vector we passed in always had a length of `1`.
So, just pass the string normally, instead of a vector.